src/corehost/cli/nethost/nethost.cpp

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "nethost.h"
+#include <error_codes.h>
+#include <fxr_resolver.h>
+#include <host_interface.h>
+#include <pal.h>
+#include <trace.h>
+#include <utils.h>
+
+NETHOST_API int NETHOST_CALLTYPE nethost_get_hostfxr_path(
+    nethost_get_hostfxr_path_result_fn result,
+    const char_t * assembly_path)
+{
+    host_mode_t mode = host_mode_t::invalid;
+    pal::string_t root_path;
+    if (assembly_path == nullptr)
+    {
+        mode = host_mode_t::muxer;
```

Comment:
Sorry - this is my bad.
We can't use the muxer behavior here. The muxer behavior assumes that the `dotnet.exe` is called and thus it searches next to it to find `hostfxr` - and nowhere else. That's effectively useless for the `nethost` scenarios, because the `nethost` will typically run in the cotext of some application, which will not have the `hostfxr` next to it.

We will need to add some more tweaks. I think the proper behavior should be:
* `assembly_path` is `nullptr`: basically the `apphost` behavior but without the "root_path", so don't look next to itself. But do use `DOTNET_ROOT` and do look in global locations.
* `assembly_path` specified: exactly like `apphost` behavior - this is already below.

Reply:
Updated to stop using the muxer behaviour and follow what you outlined above.